src/components/textfieldSlider/js/TextfieldSlider.js

```diff
@@ -66,123 +62,117 @@ var fluid_2_0 = fluid_2_0 || {};
         sliderOptions: {
             orientation: "horizontal",
             step: 1.0
-        },
-        invokers: {
-            refreshView: {
-                funcName: "fluid.textfieldSlider.refreshView",
-                args: ["{that}"]
-            }
-        },
-        finalInitFunction: "fluid.textfieldSlider.finalInit",
-        renderOnInit: true
-    });
-
-    fluid.textfieldSlider.finalInit = function (that) {
-
-        that.applier.modelChanged.addListener("value", function (newModel) {
-            that.events.modelChanged.fire(newModel.value);
-        });
-
-        if (that.options.renderOnInit) {
-            that.refreshView();
         }
-    };
-
-    fluid.textfieldSlider.refreshView = function (that) {
-        that.textfield.container.val(that.model.value);
-        that.events.afterRender.fire(that);
-    };
+    });
 
     fluid.defaults("fluid.textfieldSlider.textfield", {
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.stringToNumber",
+                input: "{that}.model.stringValue"
+            }
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            },
+            // When the input field receives a invalid input such as a string value, ignore it and populate the field with the previous value
+            stringValue: {
```

Comment:
This is a very interesting setup, that we need to talk over and write up. It depends on an odd subtlety in order to work - that an update to the value held in our model only occurs at a coarse granularity - say, whenever we receive a "change" event or some other similar checkpoint. If these updates occurred any faster, the user would be infuriated by being unable to type into the field. This is an area in which UX has spilled over into our model configuration, which to some extent we should try to prevent - especially since with our current implementation, the user could never reconfigure this listener. At the very least, we need to give a namespace to this listener - unfortunately the current dataBinding implementation will ignore it. Still, we should add the namespace with a comment that explains that having this listener here represents an interaction decision rather than an implementation decision, and that we need to be able to revisit the issue of what the behaviour is here.